### DISCUSSION TITLE
Switch between submenus

### DISCUSSION THREAD

**Discussion by grbba** [daniele77/cli#175] [Q&A]
Hello,

after a longer break i can pickup my project again and do some development :). One question for the usability of the system came up: To jump from one submenu to another without navigating through the whole menu tree.

RootMenu ----- sub A
                   |
                   ----  sub B

in sub A and in sub B i can go back to the RootMenu and then to either sub A or sub B. It would be nice if i can also ( by configuration ) add e.g. sub A to sub B so that I can jump from sub B directly to sub A without going through the RootMenu.

Is that possible ? or any workaround for that?

Thanks a lot
grbba

**Comment by daniele77** [comment]
I got your point, thank you. When the application has got many menus and submenus, one might actually desire a fastest way to navigate between menus.
However, I don't like very much the solution you're proposing. I'd rather prefer to keep the menu graph a tree (not a cyclic graph), to avoid confusion in the mind of the user (if he finds the same menu in different submenus he might not understand if they are the same menu in different places or two different menus with the same name in different places).
I'd like best to provide navigation through complete paths (as you do when you change directory in the bash) e.g.,
`/subB/subC` to go directly from whatever path to subC inside subB.
